### DISCUSSION TITLE
How to use it for local websites?

### DISCUSSION THREAD

**Discussion by LietzP** [libvips/libvips#2906] [Help wanted]
In the company I'm working for we use libvips to convert our jpg and png files to webp on all of our websites, by changing the output format of our derivatives. 

I was curious if you could do the same thing locally? 

My team leader said he never did that before, but would be curious, too.

**Comment by jcupitt** [comment]
Hi @LietzP,

Sure, you can run vips at the command-line to convert stuff (if I understand your meaning): 

```
vips copy something.jpg some-other-file-name.webp
```

The webpsave options are listed here:

https://www.libvips.org/API/current/VipsForeignSave.html#vips-webpsave

You can set them in the `[]` after the filename, eg.:

```
vips copy something.jpg some-other-file-name.webp[near-lossless,Q=40]
```

**Reply by LietzP** [reply]
Hm not exactly what I wanted.
What I thought about was: We have vips installed on our server so that all of our images (on our websites) are automatically converted to webp using derivatives. But they are still stored as *.png or *.jpg files in the backend. 
I thought about using something like that locally, too?

**Reply by jcupitt** [reply]
Do you mean your webserver converts them to webp on demand? You'd need to run a local server then I guess, and call out to libvips from that.

**Reply by LietzP** [reply]
any recommendations how I would do that?

**Reply by jcupitt** [reply]
It really depends on your platform and the detail of your server setup. I generally use apache, so I'd maybe use a docker container and clone the site into that.

**Reply by LietzP** [reply]
I'm using Preside with Lucee on Windows 10

**Reply by jcupitt** [reply]
Sorry, I've not used them. You'd need to ask on their site.

**Reply by wilspi** [reply]
@LietzP 
You can use `cwebp` to convert (compress) png/jpeg to webp locally, not sure if this is what you were looking for
https://developers.google.com/speed/webp/docs/cwebp
```
cwebp -q 80 screenshot.png -o screenshop.webp
```

**Comment by Assiente** [comment]
I think you can implement this with the script I recommended you above.
